x-pack/docs/en/rest-api/security/bulk-update-api-keys.asciidoc

```diff
@@ -2,4 +2,367 @@
 [[security-api-bulk-update-api-keys]]
 === Bulk update API keys API
 
-coming::[8.5.0]
+++++
+<titleabbrev>Bulk update API keys</titleabbrev>
+++++
+
+[[security-api-bulk-update-api-keys-request]]
+==== {api-request-title}
+
+`POST /_security/api_key/_bulk_update`
+
+[[security-api-bulk-update-api-keys-prereqs]]
+==== {api-prereq-title}
+
+* To use this API, you must have at least the `manage_own_api_key` cluster privilege.
+Users can only update API keys that they created or that were granted to them.
+To update another user's API key, use the <<run-as-privilege,`run_as` feature>>
+to submit a request on behalf of another user.
+
+IMPORTANT: It's not possible to use an API key as the authentication credential for this API.
+To update API keys, the owner user's credentials are required.
+
+[[security-api-bulk-update-api-keys-desc]]
+==== {api-description-title}
+
+This API is similar to <<security-api-update-api-key,update single API Key>> but allows you to apply the same update to multiple API keys in one API call.
+This can greatly improve performance over making individual updates.
+
+It's not possible to update expired API keys, or API keys that have been invalidated by <<security-api-invalidate-api-key,invalidate API Key>>.
+
+This API supports updates to API key access scope and metadata.
+The access scope of each API key is derived from the <<security-api-bulk-update-api-keys-api-key-role-descriptors,`role_descriptors`>> you specify in the request, and a snapshot of the owner user's permissions at the time of the request.
+The snapshot of the owner's permissions is updated automatically on every call.
+
+[IMPORTANT]
+====
+If you don't specify <<security-api-bulk-update-api-keys-api-key-role-descriptors,`role_descriptors`>> in the request, a call to this API might still change an API key's access scope.
+This change can occur if the owner user's permissions have changed since the API key was created or last modified.
+====
+
+[[security-api-bulk-update-api-keys-request-body]]
+==== {api-request-body-title}
+
+You can specify the following parameters in the request body.
+
+`ids`:: (Required, list) The IDs of the API keys to update.
+
+[[security-api-bulk-update-api-keys-api-key-role-descriptors]]
+`role_descriptors`::
+(Optional, object) The role descriptors to assign to the API keys.
+An API key's effective permissions are an intersection of its assigned privileges and the point in time snapshot of permissions of the owner user.
+If no privileges are assigned, the API key inherits the owner user's full permissions.
+You can assign new privileges by specifying them in this parameter.
+To remove assigned privileges, you can supply an empty `role_descriptors` parameter.
```

Comment:
We went over this previously. But somehow it still reads a bit confusing to me. I feel users may still be baffled with "no privileges", "remove assigned privileges" and "leave out this parameter". How about we re-arrange and adjust it a bit like the follows?
```suggestion
You can assign new privileges by specifying them in this parameter.
To remove assigned privileges, you can supply an empty `role_descriptors` parameter, i.e. an empty object `{}`.
If an API key has no assigned privileges, it inherits the owner user's full permissions.
```

Reply:
I'm good with this rephrasing. I feel like I've stared at this paragraph so much I don't know what's clear anymore 😅 but I like your proposed phrasing 👍